### ISSUE TITLE
"Hide File" context menu item does nothing

### ISSUE THREAD

**Issue by Vidar-Freespee** [JeremyFunk/hidefiles#11]
The extension hides files if they are manually added to the settings.json file, but the "Hide File" context menu item in the file browser doesn't do anything for me.

Some observations:
The command does run: if I haven't chosen local or global yet I get a toast complaining about it.
No error seems to be happening: No toasts appear and there's nothing in the extension host log (apart from the extension's activation)
It's not a refresh issue: the settings file just doesn't change.
The problem is not tied to a specific target: Nothing works, wether it's folders or files, a single item or multiple at once.

Any ideas? I'm on a Mac btw.

**Comment by JeremyFunk** [comment]
I currently don't have access to MacOS. I will test it more extensively with linux in the coming days, maybe I can replicate it there.

**Comment by ddnexus** [comment]
Same here with Ubuntu latest. It worked exactly once. Then it stopped forever. even after cleaning up the settings, uninstalling and reinstalling. It does not work.

**Comment by JeremyFunk** [comment]
Ok that is bad, I will fix it tomorrow. Can you let me know if it works once I push the update? Will @ you on this issue.

**Comment by ddnexus** [comment]
Yes, definitely. I am looking forward to see it, because it is the only extension that is actually useful for that matter.

While at it: is there any possibility to have a default setup that uses the gitignored files? That is really missing in the ecosystem. Other extensions I tried are quite messy.

Thank you.

**Comment by JeremyFunk** [comment]
It should be fixed now for both MacOS and Linux (version 2.1.2). Does it work for you?

I like the idea of linking the gitignore to Hide Files. I am very busy this week but I should find time on the weekend, I will keep you posted here.

**Comment by ddnexus** [comment]
It looks like it works. Thank you!

A few suggestions:

### Add another default with "Hidden Files"  (glob `**/.*`)

I added it by myself, but having it as default would be a nice touch for new users.

### The profiles could be more useful as composable

Instead of being one profile or the other, we could have a list of profiles and the user could select multiple profiles. 

So you could have for example: Linux-hidden + Gitignored + Configs that you can combine in all the possible permutations. Technically it would just be a matter of creating the active profile as the addition of all the selected arrays of globs.

**Comment by Vidar-Freespee** [comment]
Seems to be working. Thank you.

**Comment by JeremyFunk** [comment]
I like the idea, I will consider it when rewriting hide files in the next weeks for version 3.

**Comment by gsingh93** [comment]
The ability to add glob patterns from the context menu would be helpful. Otherwise it doesn't seem like the extension works with multi-root workspaces, as it will add the absolute path to the file which won't result in it being ignored. What needs to be added in this case would be `**/relative/path/to/file.txt`.
